Provider crashes on auth0_custom_domain data source

### Checklist

- [x] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [x] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [x] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [x] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [x] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [x] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Provider plugin crashes when trying to read data source auth0_custom_domain, but **it was not set in tenant**. Provider should return descriptive error. 


```

│ Error: Plugin did not respond
│
│   with data.auth0_custom_domain.current,
│   on data.tf line 19, in data "auth0_custom_domain" "current":
│   19: data "auth0_custom_domain" "current" {}
│
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.
╵

Stack trace from the terraform-provider-auth0_v1.15.0 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 59 :
github.com/auth0/terraform-provider-auth0/internal/auth0/customdomain.readCustomDomainForDataSource({0x105776988?, 0x140000e89a0?}, 0x140001a1400, {0x10564ad80?, 0x140005557e0?})
    github.com/auth0/terraform-provider-auth0/internal/auth0/customdomain/data_source.go:36 +0x198
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x140005f9200, {0x1057768e0, 0x140003bdcb0}, 0x140001a1400, {0x10564ad80, 0x140005557e0})
    github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:855 +0xe4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0x140005f9200, {0x1057768e0, 0x140003bdcb0}, 0x140001a1300, {0x10564ad80, 0x140005557e0})
    github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:1075 +0x110
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0x14000134000, {0x1057768e0?, 0x140003bdbf0?}, 0x140003bdb90)
    github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/grpc_provider.go:1474 +0x5a0
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0x140001761e0, {0x1057768e0?, 0x140003bd140?}, 0x14000060b40)
    github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:688 +0x1cc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x105741ce0, 0x140001761e0}, {0x1057768e0, 0x140003bd140}, 0x140001a0e00, 0x0)
    github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:665 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000262600, {0x1057768e0, 0x140003bd0e0}, 0x1400010e360, 0x14000311920, 0x105dc0b00, 0x0)
    google.golang.org/grpc@v1.69.4/server.go:1392 +0xc44
google.golang.org/grpc.(*Server).handleStream(0x14000262600, {0x105776f58, 0x140004b1520}, 0x1400010e360)
    google.golang.org/grpc@v1.69.4/server.go:1802 +0x910
google.golang.org/grpc.(*Server).serveStreams.func2.1()
    google.golang.org/grpc@v1.69.4/server.go:1030 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 71
    google.golang.org/grpc@v1.69.4/server.go:1041 +0x13c

Error: The terraform-provider-auth0_v1.15.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expectation

Provider returns error.

### Reproduction

1. do not set custom domain in tenant
2. try to use `auth0_custom_domain`

### Auth0 Terraform Provider version

1.15.0

### Terraform version

1.9.0

--- a/internal/auth0/customdomain/data_source.go
+++ b/internal/auth0/customdomain/data_source.go
@@ -2,6 +2,7 @@ package customdomain
 
 import (
 	"context"
+	"errors"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
@@ -31,6 +32,9 @@ func readCustomDomainForDataSource(ctx context.Context, data *schema.ResourceDat
 		return diag.FromErr(err)
 	}
 
+	if len(customDomains) == 0 {
+		return diag.FromErr(errors.New("no custom domain configured on tenant"))
+	}
 	// At the moment there can only ever
 	// be one custom domain configured.
 	customDomain := customDomains[0]

--- a/internal/auth0/customdomain/data_source_test.go
+++ b/internal/auth0/customdomain/data_source_test.go
@@ -2,6 +2,7 @@ package customdomain_test
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 	"testing"
 
@@ -27,6 +28,10 @@ func TestAccDataSourceCustomDomain(t *testing.T) {
 
 	acctest.Test(t, resource.TestCase{
 		Steps: []resource.TestStep{
+			{
+				Config:      `data "auth0_custom_domain" "test" {}`,
+				ExpectError: regexp.MustCompile("no custom domain configured on tenant"),
+			},
 			{
 				Config: acctest.ParseTestName(testAccDataSourceCustomDomain, testName),
 				Check: resource.ComposeTestCheckFunc(

--- a/test/data/recordings/TestAccDataSourceCustomDomain.yaml
+++ b/test/data/recordings/TestAccDataSourceCustomDomain.yaml
@@ -6,6 +6,41 @@ interactions:
         proto: HTTP/1.1
         proto_major: 1
         proto_minor: 1
+        content_length: 0
+        transfer_encoding: []
+        trailer: {}
+        host: terraform-provider-auth0-dev.eu.auth0.com
+        remote_addr: ""
+        request_uri: ""
+        body: ""
+        form: {}
+        headers:
+            Content-Type:
+                - application/json
+            User-Agent:
+                - Go-Auth0/1.19.0
+        url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains
+        method: GET
+      response:
+        proto: HTTP/2.0
+        proto_major: 2
+        proto_minor: 0
+        transfer_encoding: []
+        trailer: {}
+        content_length: 2
+        uncompressed: false
+        body: '[]'
+        headers:
+            Content-Type:
+                - application/json; charset=utf-8
+        status: 200 OK
+        code: 200
+        duration: 655.006125ms
+    - id: 1
+      request:
+        proto: HTTP/1.1
+        proto_major: 1
+        proto_minor: 1
         content_length: 133
         transfer_encoding: []
         trailer: {}
@@ -19,7 +54,7 @@ interactions:
             Content-Type:
                 - application/json
             User-Agent:
-                - Go-Auth0-SDK/0.15.1
+                - Go-Auth0/1.19.0
         url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains
         method: POST
       response:
@@ -28,71 +63,34 @@ interactions:
         proto_minor: 0
         transfer_encoding: []
         trailer: {}
-        content_length: 333
+        content_length: 443
         uncompressed: false
-        body: '{"custom_domain_id":"cd_lpxrpAKsPeWXCwAv","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"cname","record":"terraform-provider-auth0-dev-cd-lpxrpakspewxcwav.edge.tenants.eu.auth0.com"}]},"tls_policy":"recommended"}'
+        body: '{"custom_domain_id":"cd_dyVW9FYjFx7TgMz3","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"CNAME","record":"terraform-provider-auth0-dev-cd-dyvw9fyjfx7tgmz3.edge.tenants.sus.auth0.com","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com"}]},"tls_policy":"recommended"}'
         headers:
             Content-Type:
                 - application/json; charset=utf-8
         status: 201 Created
         code: 201
-        duration: 4.233335333s
-    - id: 1
-      request:
-        proto: HTTP/1.1
-        proto_major: 1
-        proto_minor: 1
-        content_length: 5
-        transfer_encoding: []
-        trailer: {}
-        host: terraform-provider-auth0-dev.eu.auth0.com
-        remote_addr: ""
-        request_uri: ""
-        body: |
-            null
-        form: {}
-        headers:
-            Content-Type:
-                - application/json
-            User-Agent:
-                - Go-Auth0-SDK/0.15.1
-        url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains/cd_lpxrpAKsPeWXCwAv
-        method: GET
-      response:
-        proto: HTTP/2.0
-        proto_major: 2
-        proto_minor: 0
-        transfer_encoding: []
-        trailer: {}
-        content_length: -1
-        uncompressed: true
-        body: '{"custom_domain_id":"cd_lpxrpAKsPeWXCwAv","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"cname","record":"terraform-provider-auth0-dev-cd-lpxrpakspewxcwav.edge.tenants.eu.auth0.com"}]},"tls_policy":"recommended"}'
-        headers:
-            Content-Type:
-                - application/json; charset=utf-8
-        status: 200 OK
-        code: 200
-        duration: 132.650458ms
+        duration: 535.47225ms
     - id: 2
       request:
         proto: HTTP/1.1
         proto_major: 1
         proto_minor: 1
-        content_length: 5
+        content_length: 0
         transfer_encoding: []
         trailer: {}
         host: terraform-provider-auth0-dev.eu.auth0.com
         remote_addr: ""
         request_uri: ""
-        body: |
-            null
+        body: ""
         form: {}
         headers:
             Content-Type:
                 - application/json
             User-Agent:
-                - Go-Auth0-SDK/0.15.1
-        url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains
+                - Go-Auth0/1.19.0
+        url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains/cd_dyVW9FYjFx7TgMz3
         method: GET
       response:
         proto: HTTP/2.0
@@ -102,32 +100,31 @@ interactions:
         trailer: {}
         content_length: -1
         uncompressed: true
-        body: '[{"custom_domain_id":"cd_lpxrpAKsPeWXCwAv","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"cname","record":"terraform-provider-auth0-dev-cd-lpxrpakspewxcwav.edge.tenants.eu.auth0.com"}]},"tls_policy":"recommended"}]'
+        body: '{"custom_domain_id":"cd_dyVW9FYjFx7TgMz3","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"CNAME","record":"terraform-provider-auth0-dev-cd-dyvw9fyjfx7tgmz3.edge.tenants.sus.auth0.com","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com"}]},"tls_policy":"recommended"}'
         headers:
             Content-Type:
                 - application/json; charset=utf-8
         status: 200 OK
         code: 200
-        duration: 114.075417ms
+        duration: 537.475917ms
     - id: 3
       request:
         proto: HTTP/1.1
         proto_major: 1
         proto_minor: 1
-        content_length: 5
+        content_length: 0
         transfer_encoding: []
         trailer: {}
         host: terraform-provider-auth0-dev.eu.auth0.com
         remote_addr: ""
         request_uri: ""
-        body: |
-            null
+        body: ""
         form: {}
         headers:
             Content-Type:
                 - application/json
             User-Agent:
-                - Go-Auth0-SDK/0.15.1
+                - Go-Auth0/1.19.0
         url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains
         method: GET
       response:
@@ -138,33 +135,32 @@ interactions:
         trailer: {}
         content_length: -1
         uncompressed: true
-        body: '[{"custom_domain_id":"cd_lpxrpAKsPeWXCwAv","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"cname","record":"terraform-provider-auth0-dev-cd-lpxrpakspewxcwav.edge.tenants.eu.auth0.com"}]},"tls_policy":"recommended"}]'
+        body: '[{"custom_domain_id":"cd_dyVW9FYjFx7TgMz3","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"CNAME","record":"terraform-provider-auth0-dev-cd-dyvw9fyjfx7tgmz3.edge.tenants.sus.auth0.com","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com"}]},"tls_policy":"recommended"}]'
         headers:
             Content-Type:
                 - application/json; charset=utf-8
         status: 200 OK
         code: 200
-        duration: 81.631667ms
+        duration: 589.235959ms
     - id: 4
       request:
         proto: HTTP/1.1
         proto_major: 1
         proto_minor: 1
-        content_length: 5
+        content_length: 0
         transfer_encoding: []
         trailer: {}
         host: terraform-provider-auth0-dev.eu.auth0.com
         remote_addr: ""
         request_uri: ""
-        body: |
-            null
+        body: ""
         form: {}
         headers:
             Content-Type:
                 - application/json
             User-Agent:
-                - Go-Auth0-SDK/0.15.1
-        url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains/cd_lpxrpAKsPeWXCwAv
+                - Go-Auth0/1.19.0
+        url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains
         method: GET
       response:
         proto: HTTP/2.0
@@ -174,33 +170,32 @@ interactions:
         trailer: {}
         content_length: -1
         uncompressed: true
-        body: '{"custom_domain_id":"cd_lpxrpAKsPeWXCwAv","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"cname","record":"terraform-provider-auth0-dev-cd-lpxrpakspewxcwav.edge.tenants.eu.auth0.com"}]},"tls_policy":"recommended"}'
+        body: '[{"custom_domain_id":"cd_dyVW9FYjFx7TgMz3","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"CNAME","record":"terraform-provider-auth0-dev-cd-dyvw9fyjfx7tgmz3.edge.tenants.sus.auth0.com","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com"}]},"tls_policy":"recommended"}]'
         headers:
             Content-Type:
                 - application/json; charset=utf-8
         status: 200 OK
         code: 200
-        duration: 87.877958ms
+        duration: 832.56575ms
     - id: 5
       request:
         proto: HTTP/1.1
         proto_major: 1
         proto_minor: 1
-        content_length: 5
+        content_length: 0
         transfer_encoding: []
         trailer: {}
         host: terraform-provider-auth0-dev.eu.auth0.com
         remote_addr: ""
         request_uri: ""
-        body: |
-            null
+        body: ""
         form: {}
         headers:
             Content-Type:
                 - application/json
             User-Agent:
-                - Go-Auth0-SDK/0.15.1
-        url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains
+                - Go-Auth0/1.19.0
+        url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains/cd_dyVW9FYjFx7TgMz3
         method: GET
       response:
         proto: HTTP/2.0
@@ -210,32 +205,31 @@ interactions:
         trailer: {}
         content_length: -1
         uncompressed: true
-        body: '[{"custom_domain_id":"cd_lpxrpAKsPeWXCwAv","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"cname","record":"terraform-provider-auth0-dev-cd-lpxrpakspewxcwav.edge.tenants.eu.auth0.com"}]},"tls_policy":"recommended"}]'
+        body: '{"custom_domain_id":"cd_dyVW9FYjFx7TgMz3","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"CNAME","record":"terraform-provider-auth0-dev-cd-dyvw9fyjfx7tgmz3.edge.tenants.sus.auth0.com","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com"}]},"tls_policy":"recommended"}'
         headers:
             Content-Type:
                 - application/json; charset=utf-8
         status: 200 OK
         code: 200
-        duration: 90.23425ms
+        duration: 4.409126083s
     - id: 6
       request:
         proto: HTTP/1.1
         proto_major: 1
         proto_minor: 1
-        content_length: 5
+        content_length: 0
         transfer_encoding: []
         trailer: {}
         host: terraform-provider-auth0-dev.eu.auth0.com
         remote_addr: ""
         request_uri: ""
-        body: |
-            null
+        body: ""
         form: {}
         headers:
             Content-Type:
                 - application/json
             User-Agent:
-                - Go-Auth0-SDK/0.15.1
+                - Go-Auth0/1.19.0
         url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains
         method: GET
       response:
@@ -246,13 +240,13 @@ interactions:
         trailer: {}
         content_length: -1
         uncompressed: true
-        body: '[{"custom_domain_id":"cd_lpxrpAKsPeWXCwAv","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"cname","record":"terraform-provider-auth0-dev-cd-lpxrpakspewxcwav.edge.tenants.eu.auth0.com"}]},"tls_policy":"recommended"}]'
+        body: '[{"custom_domain_id":"cd_dyVW9FYjFx7TgMz3","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com","primary":true,"status":"pending_verification","type":"auth0_managed_certs","verification":{"methods":[{"name":"CNAME","record":"terraform-provider-auth0-dev-cd-dyvw9fyjfx7tgmz3.edge.tenants.sus.auth0.com","domain":"testaccdatasourcecustomdomain.auth.terraform-provider-auth0.com"}]},"tls_policy":"recommended"}]'
         headers:
             Content-Type:
                 - application/json; charset=utf-8
         status: 200 OK
         code: 200
-        duration: 78.119791ms
+        duration: 459.146708ms
     - id: 7
       request:
         proto: HTTP/1.1
@@ -270,8 +264,8 @@ interactions:
             Content-Type:
                 - application/json
             User-Agent:
-                - Go-Auth0-SDK/0.15.1
-        url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains/cd_lpxrpAKsPeWXCwAv
+                - Go-Auth0/1.19.0
+        url: https://terraform-provider-auth0-dev.eu.auth0.com/api/v2/custom-domains/cd_dyVW9FYjFx7TgMz3
         method: DELETE
       response:
         proto: HTTP/2.0
@@ -287,4 +281,4 @@ interactions:
                 - application/json; charset=utf-8
         status: 204 No Content
         code: 204
-        duration: 4.3539955s
+        duration: 968.22125ms

